### PR TITLE
Use `array_is_list()` instead of `ArrayUtil::isAssoc()` where possible

### PR DESCRIPTION
**Pull request by aschempp**
I think this can be removed since we have `array_is_list` now.

**TODO after merge:**
 - [ ] ~Add a corresponding rule to `contao/contao-rector`~

### CODE REVIEW

File: core-bundle/contao/library/Contao/ArrayUtil.php [hunk 1]
```diff
@@ -64,10 +64,14 @@ public static function recursiveKeySort(array &$array): void
...
 	 * @return boolean
+	 *
+	 * @deprecated Deprecated in Contao 5.4, to be removed in Contao 6. Use the array_is_list native PHP function instead.
 	 */
 	public static function isAssoc($arrArray): bool
 	{
-		return \is_array($arrArray) && !array_is_list($arrArray);
+		trigger_deprecation('contao/core-bundle', '5.4', 'Using "%s()" has been deprecated and will no longer work in Contao 6. Use the array_is_list native PHP function instead.', __METHOD__);
+
+		return !array_is_list($arrArray);
```
Comment (leofeyer) [inline core-bundle/contao/library/Contao/ArrayUtil.php]: You must not remove the `is_array()` check:

**Review by leofeyer — CHANGES_REQUESTED** [review]

File: core-bundle/contao/library/Contao/ArrayUtil.php (hunk 1 above)
Comment (aschempp) [reply core-bundle/contao/library/Contao/ArrayUtil.php]: hmm, I somehow didn't see that. Kinda defeats the purpose of the deprecation? 🙈

Comment (fritzmg) [reply core-bundle/contao/library/Contao/ArrayUtil.php]: True, it might have some utility after all.

**Comment by aschempp** [comment]
I updated the PR to only remove unnecessary use of the helper method if the variable type is already safe (and remove a case of `is_array` check before calling the method that checks the same.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @aschempp.

**Pull request merged**